File: src/main/scala/utn/frba/iasc/utils/IdGen.scala
package utn.frba.iasc.utils

import java.text.SimpleDateFormat
import java.util.{Date, UUID}

object IdGen {
  private val sdf = new SimpleDateFormat("yyyyMMddHHmm")

  def user: String = randomId("U")

  def auction: String = randomId("A")

  def bid: String = randomId("B")

  private def randomId(prefix: String): String = s"$prefix-${sdf.format(new Date())}-${UUID.randomUUID()}"
}

File: babylon-kt/babylon-data/src/main/kotlin/utn/frba/iasc/babylon/util/IdGen.kt
package utn.frba.iasc.babylon.util

import java.text.SimpleDateFormat
import java.util.*

object IdGen {
    private val sdf = SimpleDateFormat("yyyyMMddHHmm")

    fun buyer(): String = randomId("U")

    fun auction(): String = randomId("A")

    fun bid(): String = randomId("B")

    private fun randomId(prefix: String): String = "$prefix-${sdf.format(Date())}-${UUID.randomUUID()}"
}